transformer_engine/paddle/distributed.py

```diff
@@ -13,6 +13,11 @@
 import paddle.distributed.fleet.base.topology as tp
 from paddle.distributed.fleet.meta_parallel import get_rng_state_tracker
 from paddle.distributed.fleet.layers.mpu import mp_ops
+from paddle.distributed.fleet.meta_parallel import (
+    PipelineParallel,
+    PipelineParallelMicroStepLocations,
+    register_global_pipeline_parallel_hook,
```

Comment:
I don't see `register_global_pipeline_parallel_hook` in the Paddle `develop` branch or recent releases: https://github.com/PaddlePaddle/Paddle/blob/81dbca14788bdf68de2b3f2b87f8a16aa2e84342/python/paddle/distributed/fleet/meta_parallel/__init__.py

Reply:
We should guard the import then:
```python
try:
    from paddle.distributed.fleet.meta_parallel import register_global_pipeline_parallel_hook
except ImportError:
    register_global_pipeline_parallel_hook = None
```
We should also update `register_pp_fwd_begin_hook`:
```python
def register_pp_fwd_begin_hook(forward_begin_hook):
    if register_global_pipeline_parallel_hook is not None:
        register_global_pipeline_parallel_hook(
            PipelineParallelMicroStepLocations.FORWARD_BEGIN,
            forward_begin_hook,
        )
```